shell/src/test/java/org/apache/accumulo/shell/commands/HistoryCommandTest.java

```diff
@@ -56,38 +62,41 @@ public void setUp() throws Exception {
     expect(cl.hasOption("np")).andReturn(true);
     replay(cl);
 
-    History history = new MemoryHistory();
+    History history = new DefaultHistory();
     history.add("foo");
     history.add("bar");
 
     baos = new ByteArrayOutputStream();
 
     String input = String.format("!1%n"); // Construct a platform dependent new-line
-    reader = new ConsoleReader(new ByteArrayInputStream(input.getBytes()), baos);
-    reader.setHistory(history);
+    terminal = TerminalBuilder.builder().system(false)
+        .streams(new ByteArrayInputStream(input.getBytes()), baos).build();
+    reader = LineReaderBuilder.builder().history(history).terminal(terminal).build();
```

Comment:
Is there a way to add the history directly to the terminal, and using its line reader, or do we have to create a new line reader separately from the terminal?

Reply:
The line reader is what contains the history and also contains the terminal object. Not the other way around. The line reader seems to be the higher-level object though I agree that it doesn't make the most sense. It is probably like this so it doesn't differentiate itself too much from `ConsoleReader` in jline2. 

The reader object in terminal is a bit different than a LineReader and isn't as friendly to use.

Reply:
Makes sense! Thanks for the explanation.